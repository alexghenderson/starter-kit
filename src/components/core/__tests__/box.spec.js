import React from 'react';
import enzyme from 'enzyme';
import Box from '../box';

describe('Box core component', () => {
  it('should render with default props', () => {
    const wrapper = enzyme.mount(<Box>test-content</Box>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render inline box', () => {
    const wrapper = enzyme.mount(<Box inline>test-content</Box>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render different margins and insets', () => {
    const wrapper = enzyme.mount(
      <Box vertical={-1} horizontal={2} inset={3}>
        test-content
      </Box>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render different text-aligns', () => {
    const wrapper = enzyme.mount(
      <React.Fragment>
        {['inherit', 'left', 'center', 'right'].map(align => (
          <Box align={align} key={align}>
            text-{align}
          </Box>
        ))}
      </React.Fragment>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render alternate tags', () => {
    const wrapper = enzyme.mount(<Box tag="main">text-main</Box>);
    expect(wrapper).toMatchSnapshot();
  });
});
