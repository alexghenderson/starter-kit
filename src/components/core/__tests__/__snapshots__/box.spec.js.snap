// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box core component should render alternate tags 1`] = `
.emotion-0 {
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 0px;
  text-align: inherit;
  display: block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<main
  className="emotion-0"
>
  text-main
</main>
`;

exports[`Box core component should render different margins and insets 1`] = `
.emotion-0 {
  margin-left: 8px;
  margin-right: 8px;
  margin-top: -4px;
  margin-bottom: -4px;
  padding: 16px;
  text-align: inherit;
  display: block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<div
  className="emotion-0"
>
  test-content
</div>
`;

exports[`Box core component should render different text-aligns 1`] = `
Array [
  .emotion-0 {
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 0px;
  text-align: inherit;
  display: block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<div
    className="emotion-0"
  >
    text-
    inherit
  </div>,
  .emotion-0 {
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 0px;
  text-align: left;
  display: block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<div
    className="emotion-0"
  >
    text-
    left
  </div>,
  .emotion-0 {
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 0px;
  text-align: center;
  display: block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<div
    className="emotion-0"
  >
    text-
    center
  </div>,
  .emotion-0 {
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 0px;
  text-align: right;
  display: block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<div
    className="emotion-0"
  >
    text-
    right
  </div>,
]
`;

exports[`Box core component should render inline box 1`] = `
.emotion-0 {
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 0px;
  text-align: inherit;
  display: inline-block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<div
  className="emotion-0"
>
  test-content
</div>
`;

exports[`Box core component should render with default props 1`] = `
.emotion-0 {
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 0px;
  text-align: inherit;
  display: block;
}

.emotion-0 > *:not(:last-child) {
  margin-bottom: 0px;
}

<div
  className="emotion-0"
>
  test-content
</div>
`;
